receiver/riakreceiver/scraper_test.go

```diff
@@ -90,6 +92,7 @@ func TestScaperScrape(t *testing.T) {
 		desc              string
 		setupMockClient   func(t *testing.T) client
 		expectedMetricGen func(t *testing.T) pmetric.Metrics
+		metricsEnabled    bool
```

Comment:
This is a very specific test case, so we shouldn't enhance the framework to consider it specifically. A better way is to allow each test case to provide the config it needs with a similar mechanism to `setupMockClient` or `expectedMetricsGen`.